addons/stock/tests/common.py

```diff
@@ -158,25 +178,135 @@ def setUpClass(cls):
             'location_id': cls.warehouse_1.lot_stock_id.id,
         })
 
-        # Partner
-        cls.partner_1 = cls.env['res.partner'].create({
-            'name': 'Julia Agrolait',
-            'email': 'julia@agrolait.example.com',
-        })
+        # TODO: store object and id separately (type = object, type_id = id)
+        cls.picking_type_in_obj = cls.warehouse_1.in_type_id
+        cls.picking_type_in = cls.warehouse_1.in_type_id.id
+        cls.picking_type_int_obj = cls.warehouse_1.int_type_id
+        cls.picking_type_int = cls.warehouse_1.int_type_id.id
+        cls.picking_type_out_obj = cls.warehouse_1.out_type_id
+        cls.picking_type_out = cls.warehouse_1.out_type_id.id
+        cls.warehouse_1.out_type_id.reservation_method = 'manual'
+        # cls.env['stock.picking.type'].browse(cls.picking_type_out).reservation_method = 'manual'
 
-        # Product
-        cls.product_3 = cls.env['product.product'].create({
-            'name': 'Stone',  # product_3
-            'uom_id': cls.uom_dozen.id,
-            'uom_po_id': cls.uom_dozen.id,
-        })
+        # TODO: fix supplier location
+        cls.supplier_location = cls.ModelDataObj._xmlid_to_res_id('stock.stock_location_suppliers')
+        cls.supplier_location_obj = cls.env['stock.location'].browse([cls.supplier_location])[0]
+        # TODO: store location and location.id separately
+        cls.stock_location_obj = cls.warehouse_1.lot_stock_id
+        cls.stock_location = cls.warehouse_1.lot_stock_id.id
+        cls.output_location_obj = cls.warehouse_1.wh_output_stock_loc_id
+        cls.output_location = cls.warehouse_1.wh_output_stock_loc_id.id
+        location = cls.warehouse_1.lot_stock_id
+        # location = cls.StockLocationObj.browse(cls.stock_location)
+        if not location.child_ids:
+            cls.StockLocationObj.create([{
+                'name': 'Shelf 1',
+                'location_id': location.id,
+            }, {
+                'name': 'Shelf 2',
+                'location_id': location.id,
+            }])
+
+        # Model Data
+        pack_location = cls.warehouse_1.wh_pack_stock_loc_id
+        pack_location.active = True
+        cls.pack_location = pack_location.id
+        output_location = cls.warehouse_1.wh_output_stock_loc_id
+        output_location.active = True
+        cls.output_location = output_location.id
```

Comment:
Maybe this should be in an helper (activation of pick_ship/... on a warehouse). That way we are testing in the same condition than real system